### PR TITLE
Fix tests and devtools

### PR DESCRIPTION
**Pull request by tbranyen**
This still has a long ways to go before landing, but I want to get a good idea of what kind of change impact this will make.

### CODE REVIEW

File: packages/diffhtml/lib/tree/sync.js [hunk 1]
```diff
@@ -98,7 +74,33 @@ export default function syncTree(oldTree, newTree, patches, parentTree, specialC
...
+    let retVal = fn(oldTree, newTree, keysLookup, parentTree);
+
+    // If the consumer returned a value and it doesn't equal the existing tree,
+    // then splice it into the parent (if it exists) and run a sync.
+    if (retVal && retVal !== newTree) {
+      // Synchronize this new tree.
+      newTree = retVal;
+    }
+    else if (retVal && retVal === oldTree) {
+      shortCircuit = true;
```
Comment (mAAdhaTTah) [inline packages/diffhtml/lib/tree/sync.js]: 👀

**Comment by tbranyen** [comment]
144 files changed (my lucky number) and a passed build. Job well done!

J/K, Still has a long way to go, but hell I'll take the motivation a green build gives me. So here's what's left:

- [ ] Get context working again (this time pass down to the constructors)
- [ ] Get HoC working again (with new implementation)
- [ ] DevTools should load and be reasonably functional.

**Comment by tbranyen** [comment]
Currently devtools load, but since the Web Component implementation is kind of messed up at the moment all the pages load at once :|

**Comment by tbranyen** [comment]
Gonna merge and iterate.

**Pull request merged**